src/Encoder/Parser/Parser.php

```diff
@@ -131,7 +131,7 @@ private function parseData($data)
         } elseif ($this->isRefFrame()) {
             yield $this->createReplyForRef();
         } else {
-            if (is_array($data) === true) {
+            if ((is_array($data) === true) or ($data instanceof \Traversable)) {
```

Comment:
please use `||` instead of `or`
please import \Traversable with `use` directive